diff --git a/rofi-copyq b/rofi-copyq
index 26f99fe..91e0a47 100755
--- rofi-copyq
+++ rofi-copyq
@@ -1,5 +1,3 @@
 #!/usr/bin/env python3

 # https://github.com/albertlauncher/python/blob/master/CopyQ.py
 
 import json
@@ -26,9 +24,16 @@ if __name__ == '__main__':
 
     items = []
     for json_obj in json_arr:
-        text = json_obj['text']
-        text = " ".join(filter(None, text.replace("\n", " ").split(" ")))
-        items.append(text)
+        type = json_obj['mimetypes'][0]
+        if "text" in type:
+            item = json_obj['text']
+            item = " ".join(filter(None, item.replace("\n", " ").split(" ")))
+        elif "image" in type:
+            item = "|---------<!>----------<Image>----------<!>---------|"
+        else:
+            item = "|---------<!>----------<Unkown>---------<!>---------|"
+        
+        items.append(item)
 
     title = 'rofi-copyq'
     rofi = f'rofi -dmenu -i -p {title} -format i'.split()
